### PR TITLE
fix: skip reserved keys from sync conflict

### LINKED ISSUE #1057: Invalid or corrupted pad block issue while sharing a file with a new contact
### Describe the bug

In atMospherePro when a new contact is added and a file is shared with the newly created contact, There's an Invalid or corrupted pad block issue

### Steps to reproduce

1. First , clone the code from the branch "ui_redesign" branch of atMospherePro
2. Then Run the app and add a new contact
3. And then select the newly created contact and share a file

### Expected behavior

There should not be any pad block exceptions

### Screenshots

 SEVERE|2023-06-09 16:09:01.607824|EncryptionUtil|Error while decrypting value: Invalid argument(s): Invalid or corrupted pad block
I/flutter (22587): WARNING|2023-06-09 16:09:01.611390|SyncService (@aquamarine659)|Exception occurred when setting conflict info for @banana75:shared_key@aquamarine659 | Exception: Invalid argument(s): Invalid or corrupted pad block
I/flutter (22587): FINER|2023-06-09 16:09:02.541007|SyncService (@aquamarine659)|Informing 2 listeners of SyncProgress{atSign: @aquamarine659, syncStatus: SyncStatus.success, isInitialSync: false, startedAt: 2023-06-09 10:39:00.025813Z, completedAt: 2023-06-09 10:39:02.539274Z, message: null, keyInfoList:[KeyInfo{key: senthistory_v2.mospherepro@aquamarine659, syncDirection: SyncDirection.remoteToLocal , conflictInfo: null, commitOp: CommitOp.UPDATE_ALL}, KeyInfo{key: @banana75:shared_key@aquamarine659, syncDirection: SyncDirection.remoteToLocal , conflictInfo: ConflictInfo{remoteValue: null, localValue: null, errorOrExceptionMessage: Exception occurred when setting conflict info for @banana75:shared_key@aquamarine659 | Exception: Invalid argument(s): Invalid or corrupted pad block}, commitOp: CommitOp.UPDATE_ALL}, KeyInfo{key: @banana75:shared_key@aquamarine659, syncDirection: SyncDirection.localToRemote , conflictInfo: null, commitOp: CommitOp.UPDATE_ALL}, KeyInfo{key: shared_key.banana75@aquamarine659, syncDirection: SyncDir

### Smartphones

- Device: [e.g. Samsung Galaxy]
- OS: [e.g. Android]

### Were you using an atApplication when the bug was found?

atmospherePro

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
* Fix the exception "shared_key" fails to decrypt in setConflictInfo method.
* Fix the exception "Key not found in Key store" when the uncommitted entry has a CommitOp.Delete

**- How I did it**
* For 1: Modify the IF condition, `startsWith(shared_key)` to `contains(shared_key)`
* For 2: When fetching the local value, catch KeyNotFoundException and return null.

**- How to verify it**
* For 1, Updated the existing unit test
* For 2, Added a new unit test

**- Description for the changelog**
* fix: skip reserved keys from sync conflict

### CODE REVIEW

File: packages/at_client/lib/src/service/sync_service_impl.dart [hunk 1]
```diff
@@ -572,10 +572,11 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
     final key = serverCommitEntry['atKey'];
     // publickey.<atsign>@<currentatsign> is used to store the public key of
     // other atsign. The value is not encrypted.
-    // The keys starting with publickey. and shared_key. are the reserved keys
+    // The keys starting with publickey. and keys that contain shared_key
+    // (@someone:shared_key@me, shared_key.someone@me) are the reserved keys
     // and do not require actions. Hence skipping from checking conflict resolution.
     if (key.startsWith('publickey.') ||
-        key.startsWith('shared_key.') ||
+        key.contains('shared_key') ||
```
Comment (gkc) [inline packages/at_client/lib/src/service/sync_service_impl.dart]: Let's be a bit more precise - we should match on the key looking like either `@alice:shared_key@bob` or `shared_key.alice@bob`

**Review by gkc — CHANGES_REQUESTED** [review]

File: packages/at_client/lib/src/service/sync_service_impl.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply packages/at_client/lib/src/service/sync_service_impl.dart]: Modified the conditions as `key.startsWith(RegExp('shared_key.+@.+|@.+shared_key@.+'))`

File: packages/at_client/lib/src/service/sync_service_impl.dart [hunk 2]
```diff
@@ -569,13 +569,14 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
...
+    String key = serverCommitEntry['atKey'];
     // publickey.<atsign>@<currentatsign> is used to store the public key of
     // other atsign. The value is not encrypted.
-    // The keys starting with publickey. and shared_key. are the reserved keys
+    // The keys starting with publickey. and keys that contain shared_key
+    // (@someone:shared_key@me, shared_key.someone@me) are the reserved keys
     // and do not require actions. Hence skipping from checking conflict resolution.
     if (key.startsWith('publickey.') ||
-        key.startsWith('shared_key.') ||
+        key.startsWith(RegExp('shared_key.+@.+|@.+shared_key@.+')) ||
```
Comment (gkc) [inline packages/at_client/lib/src/service/sync_service_impl.dart]: `shared_key.+@.+` will match shared_keyyyyyyy@alice
`@.+shared_key@.+` will match @ssssssshared_key:phone.wavi@alice

**Review by gkc — CHANGES_REQUESTED** [review]

File: packages/at_client/lib/src/service/sync_service_impl.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply packages/at_client/lib/src/service/sync_service_impl.dart]: Updated the Regular expression and added unit tests.

**Review by gkc — APPROVED** [review]
Thanks @sitaram-kalluri

**Pull request merged**